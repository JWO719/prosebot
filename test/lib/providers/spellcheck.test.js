const SpellCheck = require('../../../lib/providers/spellcheck')

describe('SpellCheck provider', () => {
  let provider

  beforeEach(() => {
    const map = new Map()

    const obj = {
      'filename.md': 'iam not a wurd',
      'anotherfile.md': 'Tenis is a fun sporq',
      'gibberish.md': 'Absdfsdalkjhfthjsdfsdf'
    }

    for (const key in obj) {
      map.set(key, obj[key])
    }

    provider = new SpellCheck(map)
  })

  describe('#buildResults', () => {
    it('returns the expected result', () => {
      const actual = provider.buildResults()
      expect(actual).toMatchSnapshot()
    })
  })
})
