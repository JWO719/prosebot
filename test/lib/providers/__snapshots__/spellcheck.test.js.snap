// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellCheck provider #buildReasonString returns the expected string with some corrections 1`] = `
Array [
  Object {
    "line": 1,
    "reason": "\\"iam\\" is misspelled. How about: something, anotherword",
  },
  Object {
    "line": 1,
    "reason": "\\"wurd\\" is misspelled. How about: something, anotherword",
  },
]
`;

exports[`SpellCheck provider #buildReasonString returns the expected string with zero possible corrections 1`] = `
Array [
  Object {
    "line": 1,
    "reason": "\\"iam\\" is misspelled. ",
  },
  Object {
    "line": 1,
    "reason": "\\"wurd\\" is misspelled. ",
  },
]
`;

exports[`SpellCheck provider #buildResults returns the expected result 1`] = `
Map {
  "filename.md" => Array [
    Object {
      "line": 1,
      "reason": "\\"iam\\" is misspelled. ",
    },
    Object {
      "line": 1,
      "reason": "\\"wurd\\" is misspelled. ",
    },
  ],
  "anotherfile.md" => Array [
    Object {
      "line": 1,
      "reason": "\\"Tenis\\" is misspelled. ",
    },
    Object {
      "line": 1,
      "reason": "\\"sporq\\" is misspelled. ",
    },
  ],
  "gibberish.md" => Array [
    Object {
      "line": 1,
      "reason": "\\"AbsdfsdalkjhfthjsdfsdfAbsdfsdalkjhfthjsdfsdfAbsdfsdalkjhfthjsdfsdfAbsdfsdalkjhfthjsdfsdfAbsdfsdalkjhfthjsdfsdf\\" is misspelled. ",
    },
  ],
  "technical-doc.md" => Array [],
}
`;
